lib/datadog/tracing.rb

```diff
@@ -0,0 +1,198 @@
+      #   c.sampling.default_rate = 1.0
+      #   c.use :aws
+      #   c.use :rails
+      #   c.use :sidekiq
+      #   # c.diagnostics.debug = true # Enables debug output
+      # end
+      # ```
+      #
+      # Because many configuration changes require restarting internal components,
+      # invoking {.configure} is the only safe way to change `ddtrace` configuration.
+      #
+      # Successive calls to {.configure} maintain the previous configuration values:
+      # configuration is additive between {.configure} calls.
+      #
+      # The yielded configuration `c` comes pre-populated from environment variables, if
+      # any are applicable.
+      #
+      # See {Datadog::Configuration::Settings} for all available options, defaults, and
+      # available environment variables for configuration.
+      #
+      # @yieldparam [Datadog::Configuration::Settings] c the mutable configuration object
+      # @return [void]
+      # @public_api
+      def configure(&block)
+        Datadog.configure(&block)
+      end
+
+      # (see Datadog::Tracer#active_trace)
+      # @public_api
+      def active_trace
+        tracer.active_trace
+      end
+
+      # (see Datadog::Tracer#active_span)
+      # @public_api
+      def active_span
+        tracer.active_span
+      end
+
+      # (see Datadog::TraceSegment#keep!)
+      # If no trace is active, no action is taken.
+      # @public_api
+      def keep!
+        trace = active_trace
+        active_trace.keep! if trace
+      end
+
+      # (see Datadog::TraceSegment#reject!)
+      # If no trace is active, no action is taken.
+      # @public_api
+      def reject!
+        trace = active_trace
+        active_trace.reject! if trace
+      end
+
+      # (see Datadog::Tracer#active_correlation)
+      # @public_api
+      def correlation
+        tracer.active_correlation
+      end
+
+      # Textual representation of {.correlation}, which can be
+      # added to individual log lines in order to correlate them with the active
+      # trace.
+      #
+      # Example:
+      #
+      # ```
+      # MyLogger.log("#{Datadog::Tracing.log_correlation}] My log message")
+      # # dd.env=prod dd.service=billing dd.version=13.8 dd.trace_id=545847825299552251 dd.span_id=711755234730770098 My log message
+      # ```
+      #
+      # @return [String] correlation information
+      # @public_api
+      def log_correlation
+        correlation.to_log_format
+      end
+
+      # Gracefully shuts down the tracer.
+      #
+      # The public tracing API will still respond to method calls as usual
+      # but might not internally perform the expected internal work after shutdown.
+      #
+      # This avoids errors being raised across the host application
+      # during shutdown while allowing for the graceful decommission of resources.
+      #
+      # {.shutdown!} cannot be reversed.
+      # @public_api
+      def shutdown!
+        components.shutdown!
+      end
```

Comment:
Same note as for #configure above -- this is not scoped only to tracing, so I think this is even dangerous/misleading as-is.

Reply:
See the `def configure(&block)` discussion above.

Reply:
I guess the same discussion applies here, but this one looks even more weird: if someone configures profiler through `Datadog::Tracing.configure` then that's... non-optimal, but no big impact.

But I don't think a customer doing `Datadog::Tracing.shutdown!` or `Datadog::Profiling.shutdown!` really expects to stop everything in the product. This seems like really really sharp edge.

Reply:
For consistency, I think this should follow the same decision we have for `#configure`, because
 `#shutdown!` and `#configure` are effectively `components.shutdown!` and `components.shutdown! + components.startup!`, respectively.

Reply:
I changed this behavior to do `tracer.shutdown!` instead, such that it doesn't shut down all Datadog components in the application. Hopefully this is a reasonable compromise.